### DISCUSSION TITLE
How to perform MiXCR on my spatial TCR data

### DISCUSSION THREAD

**Discussion by yuyuleung** [milaboratory/mixcr#1683] [Q&A] [answered]
Hello, 

I wanna perform MiXCR analysis on my spatial TCR data, where multiple barcodes are assigned to a spatial spot (looked as a single cell). 

My .fq files are in the following format:
read1.fq: 20bp (barcode)
read2.fq: 90bp (mRNA)

Besides, I have also generated a barcodeTranslate file, where which barcodes can be assigned to which cells are clarified. 
Like:
spot1 CAACAAACAAAAATACTATATTAAC
spot1 CAACAAACATAAATACAATATTAAC
spot2 TACAACTGAAAATAATTATCATACC
spot2 TACAACTGAATATAATTATCATACG
...

How can I specify/define my multiple barcode sequences as a single cell to analyze? 

Thank you. 
Yuyu

**Comment by mizraelson** [comment]
Hi, what protocol did you use? Is it 10x Visium?

**Reply by yuyuleung** [reply]
Hello, thanks for asking. 

I am not really using 10x visium, but a protocol like it. The library dose not contain UMI but only barcode. Cause a single barcode is relative smaller than a cell, I wanna merge several barcodes closing together as a spot and assemble reads from them. Do you have any idea, how I can perform my spatial data with MiXCR?

Thanks
Yuyu

**Reply by mizraelson** [reply, marked as answer]
Hi,
Assuming your protocol is also based on fragmentation, similar to 10x, and you expect each spot (which contains multiple barcodes) to yield a single cell, I would recommend starting with [the custom preset](https://github.com/user-attachments/files/15795720/preset.zip). Open the YAML file in any text editor and add your barcodes, similar to the example shown there, based on the two-spot barcodes you have shared. Then, place it in the directory from which you run MiXCR, or in the `~/.mixcr/presets` folder. The command would then be:

```
mixcr analyze local:spatial-tcr-1683 \
    --species hsa \
    input_R1.fastq.gz \
    input_R1.fastq.gz \
    output
```

**Reply by yuyuleung** [reply]
Thank you so much for your tips. I will try on it. :)

**Reply by mizraelson** [reply]
Of course! Feel free to reach should you have any questions. You can also write us at support@milaboratories.com
